core/include/system.h

```diff
@@ -76,6 +82,14 @@ class System {
      */
     uint64_t get_uuid() const;
 
+    /**
+     * @brief Register a callback to be called when a component is discovered.
+     *
+     * @param callback a function of type void(uint8_t) which will be called with the COMPONENT_ID
```

Comment:
I would prefer if we don't "leak" the mavlink component ID outside of the SDK but only the notion of autopilot/camera/gimbal. Therefore, in my opinion, it should be something like `register_camera_discovered()`.

Reply:
Is there a convention on how components should be called? I mean how do you know, from MAVLink, that it is a camera?

Reply:
The camera has `MAV_COMP_ID_CAMERA[x]`, it's something known to the SDK internals.

https://mavlink.io/en/messages/common.html#MAV_COMP_ID_CAMERA

Reply:
@julianoes Can we declare an enum like ::Camera ::Gimbal ::AutoPilot ::Unknown etc (maybe this already exists and pass that to the callback? that way there'd only be one callback function instead of one function per component.

Reply:
Yes an enum for components makes sense, I agree.

Reply:
@julianoes change pushed.

Reply:
Cool, the comment needs to be adapted though.